src/database/statistics/dateUtil.ts

```diff
@@ -85,62 +56,158 @@ export function spreadsGroupedByTimeslots(timeRepoPairs: {
     sprints: 0,
     months: 0,
   };
-  // init first dayspread entry
-  dates.daySpread[timestamps[0]] = timeRepoPairs[timestamps[0]];
-  // init all category counters with the first timestamp
+
+  // values to compare with next timestamp in the loop
   let weekDate = timestamps[0];
-  let sprintDate = timestamps[0];
   let monthDate = timestamps[0];
-  // counter for category sums
-  let daySpreadSum = timeRepoPairs[timestamps[0]].length; // already init it with the length of the first entrys repoID array
+
+  // initialize all spread objects with the first timestamp and related repoID arr,
+  // as the first timestamp relates to first day, first week and first month at all.
+  dates.daySpread[timestamps[0]] = timeRepoPairs[timestamps[0]];
+  const week = getCalendarWeek(weekDate);
+  dates.weekSpread[week] = timeRepoPairs[timestamps[0]];
+  const month = new Date(monthDate).getMonth() + 1;
```

Comment:
Since you are adding 1 to the month's index, I would extract this into a `getMonth` method, especially since you do this at least 3 times.

Reply:
I have outsourced it to `getMonthNumber()`